aqt/metadata.py

```diff
@@ -276,25 +320,33 @@ def to_url(self) -> str:
 
     def to_folder(self, version: Version, qt_version_no_dots: str, extension: Optional[str] = None) -> str:
         if version >= Version("6.8.0"):
-            return "{category}{major}_{ver}/{category}{major}_{ver}{ext}".format(
-                category=self.category,
-                major=qt_version_no_dots[0],
-                ver=qt_version_no_dots,
-                ext="_" + extension if extension else "",
-            )
-        else:
-            return "{category}{major}_{ver}{ext}".format(
-                category=self.category,
-                major=qt_version_no_dots[0],
-                ver=qt_version_no_dots,
-                ext="_" + extension if extension else "",
-            )
+            if self.target == "wasm":
+                # Qt 6.8+ WASM uses a split folder structure
+                folder = f"qt{version.major}_{qt_version_no_dots}"
+                if extension:
+                    folder = f"{folder}/{folder}_{extension}"
+                return folder
+            else:
+                base = f"qt{version.major}_{qt_version_no_dots}"
+                return f"{base}/{base}"
+        elif version >= Version("6.5.0") and self.target == "wasm":
+            # Qt 6.5-6.7 WASM uses direct wasm_[single|multi]thread folder
+            if extension:
+                return f"qt{version.major}_{qt_version_no_dots}_{extension}"
+            return f"qt{version.major}_{qt_version_no_dots}"
+        # Pre-6.8 structure for non-WASM or pre-6.5 structure
+        return "{category}{major}_{ver}{ext}".format(
+            category=self.category,
```

Comment:
The category handling is a bit different for >= 6.8.0 non-wasm, category is now assumed to be qt in that case.  category is either tools or qt.  Is there any case where the directory component is tools{major}_{ver}{ext}, either before or after 6.8.0?

Reply:
This return is there for the legacy versions. For Qt 6.8 non-WASM, this is returned:
```python
base = f"qt{version.major}_{qt_version_no_dots}"
return f"{base}/{base}"
```

All tests pass with this `to_folder`, older and newer versions of Qt, WASM of not